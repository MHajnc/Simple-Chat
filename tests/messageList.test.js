import React from 'react';
import renderer from 'react-test-renderer';

import MessageList from '../app/components/chat/MessageList';

test('Testing MessageList Component', () => {
  const messages = [
    {currentuser: true, text: "Hello!", type: "message"},
    {currentuser: false, text: "Hi there...", type: "message"},
    {currentuser: true, text: "Do you want to talk?", type: "message"},
    {currentuser: false, text: "NO!!!", type: "message"},
    {status: 'someone left', count: 1, type: 'status'}
  ];

  const out = renderer.create(
    <MessageList messagelist={messages} />
  ).toJSON();

  expect(out).toMatchSnapshot();
});
