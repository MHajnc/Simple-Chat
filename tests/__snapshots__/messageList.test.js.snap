// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MessageList Component 1`] = `
<div
  className="chat__messageList"
  id="messageList"
>
  <div
    className="chat__messageList__userMessage"
  >
    <div
      className="chat__messageList__userText"
    >
       
      Hello!
       
    </div>
  </div>
  <div
    className="chat__messageList__incomingMessage"
  >
    <div
      className="chat__messageList__messageText"
    >
       
      Hi there...
       
    </div>
  </div>
  <div
    className="chat__messageList__userMessage"
  >
    <div
      className="chat__messageList__userText"
    >
       
      Do you want to talk?
       
    </div>
  </div>
  <div
    className="chat__messageList__incomingMessage"
  >
    <div
      className="chat__messageList__messageText"
    >
       
      NO!!!
       
    </div>
  </div>
  <div
    className="chat__messageList__statusBox"
  >
    someone left
     - there 
    is
     
    1
     participant
  </div>
</div>
`;
